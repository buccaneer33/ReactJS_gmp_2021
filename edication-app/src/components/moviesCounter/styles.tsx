import {createUseStyles} from 'react-jss'

export const useStyles = createUseStyles({
    block: {
        width: '100%',
        color: '#FFFFFF'
    }
});
