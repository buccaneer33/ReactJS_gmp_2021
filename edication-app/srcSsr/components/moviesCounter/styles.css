.block {
    width: 100%;
    color: #FFFFFF;
}
