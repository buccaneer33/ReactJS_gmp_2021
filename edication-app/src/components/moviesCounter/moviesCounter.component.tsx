import React from 'react';
import {useStyles} from './styles';
import { MoviesArray } from '../../common/interfaces/ApiDataInterface';

export const MoviesCounterComponent = (props:MoviesArray): JSX.Element => {
    const classes = useStyles();
    return (
        <div className={classes.block}>{props.moviesInfo.length}<span> movies found</span></div>
    );
}
