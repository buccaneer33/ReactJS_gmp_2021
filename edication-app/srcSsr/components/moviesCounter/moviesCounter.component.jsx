import React, {useState} from 'react';
import styles from './styles.css';

export const MoviesCounterComponent = ({moviesCount}) => {
    const [count] = useState(moviesCount);

    return (<div class={styles.block}>{count}<span> movies found</span></div>);
}
