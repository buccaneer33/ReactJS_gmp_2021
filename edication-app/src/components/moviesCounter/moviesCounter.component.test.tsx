import React from 'react';
import {render, screen } from '@testing-library/react';
import {MoviesCounterComponent} from './moviesCounter.component';

describe('LogoComponent', () => {
  
    test('component is created', () => {
        render(<MoviesCounterComponent moviesCount={0} />);
        expect(screen).toBeDefined();
    });
    test('component conteins "100"', () => {
        render(<MoviesCounterComponent moviesCount={100} />);
        expect(screen.getByText('100')).toBeInTheDocument();
    });
    test('component conteins "movies found"', () => {
        render(<MoviesCounterComponent moviesCount={100} />);
        expect(screen.getByText('movies found')).toBeInTheDocument();
    });
    test('test snapchot component', () => {
        const component = render(<MoviesCounterComponent moviesCount={0} />).container;
        expect(component).toMatchSnapshot();
    });
});
