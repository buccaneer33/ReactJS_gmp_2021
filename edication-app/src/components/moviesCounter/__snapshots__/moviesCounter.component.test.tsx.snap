// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoComponent test snapchot component 1`] = `
<div>
  <div
    class="block-0-2-1"
  >
    0
    <span>
       movies found
    </span>
  </div>
</div>
`;
